### ISSUE TITLE
Dynamic tokens support for publishing

### ISSUE THREAD

**Issue by alexcos20** [oceanprotocol/market#1335]
Right now, we hardcode OCEAN token address in config.
Also, the publish form only supports one token.

Proposal:
- on publish, fetch approved tokens from subgraph:
```
query {
 opcs{
  approvedTokens
}
 
}
```

result:
```
{
  "data": {
    "opcs": [
      {
        "approvedTokens": [
          "0x8967bcf84170c91b0d24d4302c2376283b0b3a07"
        ]
      }
    ]
  }
}
```

- then allow publishers to use any of those approved tokens when publishing assets

**Comment by mihaisc** [comment]
Sorry, we decided that this will be after the launch, so i removed the milestone.

**Comment by alexcos20** [comment]
Right. it's low priority, just in case somebody has nothing else to do :)

**Comment by kremalicious** [comment]
there's a bit more to this than just publish form, #468 needs to be finalized first or done together with this here. In any case, huge undertaking so nobody will do this on the side.

**Comment by LucaMilanese90** [comment]
Quick question: I see that with the query provided by Alex the result is always an array with one token (except for the ETH mainnet), how can I get a token address to run some tests and see if everything works as intended?

**Comment by mihaisc** [comment]
@alexcos20 can we add a second approved token on a test network?

**Comment by alexcos20** [comment]
Rinkeby:    [0x8967bcf84170c91b0d24d4302c2376283b0b3a07](https://rinkeby.etherscan.io/address/0x8967bcf84170c91b0d24d4302c2376283b0b3a07), [0xd92e713d051c37ebb2561803a3b5fbabc4962431](https://rinkeby.etherscan.io/address/0xd92e713d051c37ebb2561803a3b5fbabc4962431) address[]

Faucet: https://bybit-exchange.github.io/erc20-faucet/

**Comment by LucaMilanese90** [comment]
Thanks for the quick answer! 😄

**Comment by LucaMilanese90** [comment]
I'm having a few issues with publishing an asset with the custom base token you added on Rinkeby: 
- the TUSDT has 6 decimals instead of 18, is this a valid possibility? If so, is there an easy way to get the token info like symbol and decimals from the address in the publish phase? 
- the publish works with fixed price updating the decimals value to 6, this is not the same for a pool price, is there something else I have to change to make it work? I keep getting [transactions failing](https://rinkeby.etherscan.io/tx/0x9d636704aa992d720d6f3d72ccaf7a0d7314120393726980135634c235a3c360) this errors 
![Screenshot 2022-06-13 at 17 28 18](https://user-images.githubusercontent.com/20192135/173389432-ca9d7c07-6d8a-46f6-ac2f-edbdd5a27feb.png)

**Comment by mihaisc** [comment]
already answered on the pr 😄

**Comment by LucaMilanese90** [comment]
Perfect, thanks😁
